src/python/bezier/_symbolic.py

```diff
@@ -24,33 +24,32 @@
 
 import functools
 
-try:
-    import sympy
-except ImportError:  # pragma: NO COVER
-    sympy = None
-
 
 def require_sympy(wrapped):
     """Function decorator to require ``sympy`` to exist.
 
     Args:
         wrapped (Callable): A function to be wrapped.
+            The function should take ``sympy`` as its first parameter; the
+            decorator will prepend the module to the actual parameter list.
 
     Returns:
         Callable: The wrapped function.
     """
 
     @functools.wraps(wrapped)
     def ensure_sympy(*args, **kwargs):
-        if sympy is None:
+        try:
+            import sympy  # pylint: disable=import-outside-toplevel
+        except ImportError:  # pragma: NO COVER
             raise OSError("This function requires SymPy.")
-        return wrapped(*args, **kwargs)
+        return wrapped(sympy, *args, **kwargs)
 
     return ensure_sympy
 
 
 @require_sympy
-def to_symbolic(nodes):
+def to_symbolic(sympy, nodes):
```

Comment:
This would mean `Args` needs to be updated for this function. (Once upon a time Pylint would've flagged the mismatch, not sure what the difference is.)

Reply:
Ah I spoke too soon, [lint failure][1] from this PR:

```
************* Module bezier._symbolic
src/python/bezier/_symbolic.py:52:0: W9015: "sympy" missing in parameter documentation (missing-param-doc)
src/python/bezier/_symbolic.py:52:0: W9016: "sympy" missing in parameter type documentation (missing-type-doc)
src/python/bezier/_symbolic.py:73:0: W9015: "sympy" missing in parameter documentation (missing-param-doc)
src/python/bezier/_symbolic.py:73:0: W9016: "sympy" missing in parameter type documentation (missing-type-doc)
src/python/bezier/_symbolic.py:98:0: W9015: "sympy" missing in parameter documentation (missing-param-doc)
src/python/bezier/_symbolic.py:98:0: W9016: "sympy" missing in parameter type documentation (missing-type-doc)
src/python/bezier/_symbolic.py:111:16: E1120: No value for argument 'nodes' in function call (no-value-for-parameter)
src/python/bezier/_symbolic.py:114:31: E1120: No value for argument 's' in function call (no-value-for-parameter)
src/python/bezier/_symbolic.py:122:0: W9015: "sympy" missing in parameter documentation (missing-param-doc)
src/python/bezier/_symbolic.py:122:0: W9016: "sympy" missing in parameter type documentation (missing-type-doc)
src/python/bezier/_symbolic.py:139:0: W9015: "sympy" missing in parameter documentation (missing-param-doc)
src/python/bezier/_symbolic.py:139:0: W9016: "sympy" missing in parameter type documentation (missing-type-doc)
src/python/bezier/_symbolic.py:157:22: E1120: No value for argument 'degree' in function call (no-value-for-parameter)
src/python/bezier/_symbolic.py:159:11: E1120: No value for argument 's' in function call (no-value-for-parameter)
src/python/bezier/_symbolic.py:139:22: W0613: Unused argument 'sympy' (unused-argument)
src/python/bezier/_symbolic.py:163:0: W9015: "sympy" missing in parameter documentation (missing-param-doc)
src/python/bezier/_symbolic.py:163:0: W9016: "sympy" missing in parameter type documentation (missing-type-doc)
src/python/bezier/_symbolic.py:197:0: W9015: "sympy" missing in parameter documentation (missing-param-doc)
src/python/bezier/_symbolic.py:197:0: W9016: "sympy" missing in parameter type documentation (missing-type-doc)
src/python/bezier/_symbolic.py:211:16: E1120: No value for argument 'nodes' in function call (no-value-for-parameter)
src/python/bezier/_symbolic.py:214:31: E1120: No value for argument 't' in function call (no-value-for-parameter)
src/python/bezier/_symbolic.py:222:0: W9015: "sympy" missing in parameter documentation (missing-param-doc)
src/python/bezier/_symbolic.py:222:0: W9016: "sympy" missing in parameter type documentation (missing-type-doc)
src/python/bezier/_symbolic.py:246:0: W9015: "sympy" missing in parameter documentation (missing-param-doc)
src/python/bezier/_symbolic.py:246:0: W9016: "sympy" missing in parameter type documentation (missing-type-doc)
src/python/bezier/_symbolic.py:264:25: E1120: No value for argument 'degree' in function call (no-value-for-parameter)
src/python/bezier/_symbolic.py:266:11: E1120: No value for argument 't' in function call (no-value-for-parameter)
src/python/bezier/_symbolic.py:246:25: W0613: Unused argument 'sympy' (unused-argument)
************* Module bezier.triangle
src/python/bezier/triangle.py:1180:29: E1120: No value for argument 'degree' in function call (no-value-for-parameter)
src/python/bezier/triangle.py:1218:15: E1120: No value for argument 'degree' in function call (no-value-for-parameter)
************* Module bezier.curve
src/python/bezier/curve.py:764:26: E1120: No value for argument 'degree' in function call (no-value-for-parameter)
src/python/bezier/curve.py:800:15: E1120: No value for argument 'degree' in function call (no-value-for-parameter)
```

[1]: https://app.circleci.com/jobs/github/dhermes/bezier/1692/parallel-runs/0/steps/0-112

Reply:
Well, it's not *really* an argument -- callers don't need to pass it, the decorator handles it.

I guess getting rid of `requires_sympy` and having something like
```
def _load_sympy():
    try: import sympy
    except ImportError: raise OSError(...)
    else: return sympy

def to_symbolic(nodes):
    sympy = _load_sympy()
    ...
```
would minimize repetition which remaining linter-friendly?